tensorboard/webapp/persistent_settings/_data_source/persistent_settings_data_source.ts

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Injectable} from '@angular/core';
+import {EMPTY, Observable, of} from 'rxjs';
+import {map, tap} from 'rxjs/operators';
+
+import {LocalStorage} from '../../util/local_storage';
+import {BackendSettings, PersistableSettings} from './types';
+
+const LEGACY_METRICS_LOCAL_STORAGE_KEY = '_tb_global_settings.timeseries';
+const GLOBAL_LOCAL_STORAGE_KEY = '_tb_global_settings';
+
+@Injectable()
+export abstract class PersistentSettingsDataSource<UiSettings> {
+  abstract setSettings(partialSetting: Partial<UiSettings>): Observable<void>;
+  abstract getSettings(): Observable<Partial<UiSettings>>;
+}
+
+@Injectable()
+export abstract class SettingsCoverter<UiSettings, StorageSettings> {
+  abstract uiToBackend(
+    uiSettings: Partial<UiSettings>
+  ): Partial<StorageSettings>;
+  abstract backendToUi(
+    backendSettings: Partial<StorageSettings>
+  ): Partial<UiSettings>;
+}
+
+@Injectable()
+export class OSSSettingsConverter extends SettingsCoverter<
```

Comment:
the naming`OSSSettingsConverter` makes me wonder: Is there any other converter in your mind that it also extends `SettingsCoverter`?

Reply:
Yes, as Erik asked, internal versions like TB.dev will require more settings to be persisted on top of OSS ones and this is recognizing that.